Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -728,9 +842,11 @@ private final class ResolvedPackageBuilder: ResolvedBuilder<ResolvedPackage> {
     /// The dependencies of this package.
     var dependencies: [ResolvedPackageBuilder] = []
 
-    let isAllowedToVendUnsafeProducts: Bool
+    /// The defaultLocalization for this package.
+    var defaultLocalization: String? = nil
 
-    let allowedToOverride: Bool
+    /// The platforms supported by this package.
+    var platforms: SupportedPlatforms = .init(declared: [], derived: [])
```

Comment:
I think we should continue to only expose platforms at the target level in the model. I realize we have a couple of existing places where we are using the "first" target to approximate this already, but I think that's preferable to having two distinct attributes in the model itself.

Reply:
Using `targets[0]` isn't really correct though — to really support this for each target we'd want to do the consistency checking, or have it use the newest of the various target versions, etc.  I do agree that eventually the manifest should support it for each target, but until then it seems right to match the conceptual model by having this at the package level.

Reply:
One drawback is of course that it least to more "find the package for this target" lookups, since traversal of the graph tends to be target-centric.  But it still seems conceptually more correct.

Reply:
we _are_ still exposing platforms on the target but not at the model's Target, instead we do it at the ResolvedTarget level and at the ResolvedPackage level

Reply:
> instead we do it at the ResolvedTarget level (which is a better home for derived attributes)

I agree, and I suspect we'll have more derived attributes there over time.

> also at the ResolvedPackage level in case you want to access it from that level instead of the targets[0] which is less accurate

Based on the previous comment in this thread, perhaps it should be on ResolvedProduct and not ResolvedPackage, if it never makes sense to talk about it as a whole.  I'll have to look closer but I think all the need to access this is either from a target or a product, but not the package as a whole (i.e. you never say `package.targets[0]`).

Reply:
the raw information is currently set on package, so imo it makes sense to _also_ expose it there

but happy to also expose it at product level - it will be a merge of the targets for that products iiuc?

Reply:
@neonichu @abertelrud updated this PR to compute these for ResolvedProduct as well

Reply:
> the raw information is currently set at the package, so imo it makes sense to _also_ expose it there.

That makes sense — it should ideally never be used from there (since the answer is ambiguous, e.g. tests vs not tests).  But if it's defined in the model at that layer then it makes sense to have it in `ResolvedPackage` as well.

> but happy to also expose it at product level - it will be a merge of the targets for that products iiuc?

Yes, in theory it seems like it would be the newest version of all the targets.  Right now there's no way to specify it per-target, so they would probably all be the same version (unless the targets can come from a different package — I'd have to look into whether they can).

> @neonichu @abertelrud updated this PR to compute these for ResolvedProduct as well: https://github.com/apple/swift-package-manager/pull/4191/files#diff- 6b261f02691e503e95d595d2ab260616242dbb7224439ba5d313a95d2481e7d2R93

Thanks, will take a look at this!